pkg/option/features.go

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package option
+
+import "cmp"
+
+// NetworkPolicyEnabled returns true if the network policy enforcement
+// system is enabled for K8s, Cilium and Cilium Clusterwide network policies.
+func NetworkPolicyEnabled(cfg *DaemonConfig) bool {
+	return cmp.Or(
+		cfg.EnablePolicy != NeverEnforce,
+		cfg.EnableK8sNetworkPolicy,
```

Comment:
Since this is called from the operator, we should bind `EnablePolicy` and `EnableK8sNetworkPolicy` too to the operator `Viper` instance, as you already did for `EnableCiliumNetworkPolicy` and `EnableCiliumClusterwideNetworkPolicy` in https://github.com/cilium/cilium/pull/35049

Reply:
Done.

Reply:
I see you've added `EnableK8sNetworkPolicy`, but shouldn't we bind `EnablePolicy` too?

Reply:
It's not needed, unless we want to override it additionally, but that adds more complexity.
`EnablePolicy` is derived from other config, as mentioned in the PR description.

Conditions for policy disabled:
- EnablePolicy=never
- DisableCiliumEndpointCRD=true
- EnableK8sNetworkPolicy=false
- EnableCiliumNetworkPolicy=false
- EnableCiliumClusterwideNetworkPolicy=false
- IdentityAllocationMode=crd